### ISSUE TITLE
Pass the value of project.links into the build script

### ISSUE THREAD

**Issue by mzabaluev** [rust-lang/cargo#1220]
It would be nice if the project's `links` value was available to the build script.
Then build helpers like [pkg-config](https://github.com/alexcrichton/pkg-config-rs) could read that, so that `build.rs` could be a drop-in file suitable for all -sys projects where the library provides a pkg-config file with the same name.

**Comment by mzabaluev** [comment]
Cc @alexcrichton

**Comment by alexcrichton** [comment]
Sounds like a good idea to me! Passing something like `LIB` or some other environment variable to the build script should work ok.

**Comment by sbeckeriv** [comment]
I did it backwards in my pull request. I was setting the links based on a variable. I am told I need to set the environment variable based on the links value. I am looking into setting the environment variable now. It looks like the standard lib https://doc.rust-lang.org/std/env/fn.set_var.html  only sets the variable for the current running process. Which matches up with what I am seeing building https://github.com/PistonDevelopers/freetype-sys

Also I noticed pistion build parses 3 mainifest files but the last links is different. Im guessing you would only care about the first parsed.
`
/Users/becker/trash/cargo-becker/target/debug/cargo  build
setting RUST_LIB freetype
setting RUST_LIB freetype
setting RUST_LIB z`

Let me know if there is a known way to leak out environment variable changes.
